crypto/src/main/scala/org/bitcoins/crypto/ECKey.scala

```diff
@@ -4,18 +4,143 @@ import scodec.bits.ByteVector
 
 import java.math.BigInteger
 import scala.annotation.tailrec
-import scala.concurrent.ExecutionContext
+
+/** Represents the raw bytes which are meant to represent an ECKey without deserializing. */
+sealed trait ECKeyBytes extends NetworkElement
+
+/** Represents a serialization sensitive ECPrivateKey (such as is used in WIF). */
+case class ECPrivateKeyBytes(bytes: ByteVector, isCompressed: Boolean = true)
+    extends ECKeyBytes
+    with MaskedToString {
+  val toPrivateKey: ECPrivateKey = ECPrivateKey(bytes)
+
+  /** Returns the raw ECPublicKeyBytes serialized using isCompressed. */
+  def publicKeyBytes: ECPublicKeyBytes = {
+    val pubKey = toPrivateKey.publicKey
+    if (isCompressed) {
+      ECPublicKeyBytes(pubKey.bytes)
+    } else {
+      ECPublicKeyBytes(pubKey.decompressedBytes)
+    }
+  }
+
```

Comment:
`def toPublicKey` as well ? I don't see a reason not to

Reply:
I'm kind of afraid of adding this as I can imagine people using this function when they mean to use `toPublicKeyBytes`, I think it is nice to avoid that footgun by explictly requiring `bytes.toPrivateKey.toPublicKey`if this is what they actually want.